### PR TITLE
[NetAppFiles] Apply 2022-05-01

### PR DESCRIPTION
**Pull request by audunn**
**Related command**
Update azure-mgmt-netapp dependency to version 9.0.0 and changes made according to that version.

**Description**
Update ANF Cli to api-version 2022-05-01

**Testing Guide**

**History Notes**
[NetAppFiles] `az netappfiles account create`: Add optional parameters `--key-name`, `--key-source`, `--keyvault-resource-id`, `--user-assigned-identity`
[NetAppFiles] `az netappfiles account update`: Add optional parameters `--key-name`, `--key-source`, `--keyvault-resource-id`, `--user-assigned-identity`
[NetAppFiles] `az netappfiles volume create`: Add optional parameters `--smb-access-based-enumeration`, `--smb-non-browsable`, `--delete-base-snapshot`
[NetAppFiles] `az netappfiles resource`: Add new command `query-region-info`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
NetAppFiles

**Comment by audunn** [comment]
I am not sure why (Test Extensions Loading Python310) is failing. Can someone take a look

the error I see is ModuleNotFoundError: No module named 'azext_reservation.vendored_sdks'
 

FYI some very important customers are waiting for this PR, can this make the Sep 2022 (2022-10-12) milestone that has code complete date 10/08/2022

**Comment by yonzhan** [comment]
Add @kairu-ms to take a look after National holiday

**Comment by wangzelin007** [comment]
Hi @audunn,
CI fails because pr Azure/azure-cli-extensions#5408 is missing a file `__init__.py` in `src/reservation/azext_reservation/vendored_sdks` directory and has been fixed by this pr https://github.com/Azure/azure-cli-extensions/pull/5421

File: src/azure-cli/azure/cli/command_modules/netappfiles/commands.py [hunk 1]
```diff
@@ -371,3 +379,8 @@ def load_volume_groups_command_groups(self, netappfiles_volume_groups_sdk):
                          exception_handler=netappfiles_exception_handler)
         g.command('delete', 'begin_delete', confirmation=True, supports_no_wait=True)
         g.wait_command('wait')
+
+
+def load_net_app_resource_command_groups(self, netappfiles_resource_sdk):
+    with self.command_group('netappfiles resource', netappfiles_resource_sdk) as g:
+        g.command('region-info', 'query_region_info', supports_no_wait=True)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/netappfiles/commands.py:386]: Command name should start with verb

**Comment by audunn** [comment]
@kairu-ms PR updated, thanks

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**